api/src/main/java/org/openmrs/validator/ObsValidator.java

```diff
@@ -223,6 +225,12 @@ else if (!obs.hasGroupMembers()) {
 				errors.rejectValue("concept", "must have a datatype");
 			}
 		}
+
+		if (obs.getPerson() != null && obs.getEncounter() != null && obs.getEncounter().getPatient() != null) {
+			if (Objects.equals( obs.getPerson(), obs.getEncounter().getPatient() )) {
+				errors.rejectValue( "person", "error.mismatch" );
```

Comment:
Is there actually a translation for `error.mismatch`? If not, you need to add it

Reply:
I guess no..Could you please tell me how it's done

Reply:
@wluyima I added the error into translation files. Please feel free to review..Thanks..
https://github.com/openmrs/openmrs-core/pull/2291

Reply:
As I mentioned earlier, you don't have to create a new PR when addressing review comments if the original OR is still open.

Reply:
@wluyima I had to delete the branch due to wrong push..sry...Next time I'd revert such pushes. Thanks